src/Microsoft.QualityGuidelines.Analyzers/Core/SealMethodsThatSatisfyPrivateInterfaces.Fixer.cs

```diff
@@ -22,15 +23,37 @@ public abstract class SealMethodsThatSatisfyPrivateInterfacesFixer : CodeFixProv
 
         public sealed override FixAllProvider GetFixAllProvider()
         {
-            // See https://github.com/dotnet/roslyn/blob/master/docs/analyzers/FixAllProvider.md for more information on Fix All Providers
             return WellKnownFixAllProviders.BatchFixer;
         }
 
         public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
-        {                              
-            // This is to get rid of warning CS1998, please remove when implementing this analyzer
-            await new Task(() => { });
-            throw new NotImplementedException();
+        {
+            var model = await context.Document.GetSemanticModelAsync(context.CancellationToken);
+            var gen = SyntaxGenerator.GetGenerator(context.Document);
+
+            foreach (var dx in context.Diagnostics)
+            {
+                if (dx.Location.IsInSource)
+                {
+                    var root = dx.Location.SourceTree.GetRoot(context.CancellationToken);
+                    var declarationNode = gen.GetDeclaration(root.FindToken(dx.Location.SourceSpan.Start).Parent);
+                    if (declarationNode != null)
+                    {
+                        var symbol = model.GetDeclaredSymbol(declarationNode, context.CancellationToken) as IMethodSymbol;
+                        if (symbol != null && symbol.IsOverride)
+                        {
+                            context.RegisterCodeFix(CodeAction.Create("Seal overridden method", c => SealOverriddenMethodAsync(context.Document, root, declarationNode, c)), dx);
```

Comment:
We generally create a simple CodeAction that derives from AnalyzerUtilities.DocumentChangeAction with a more specific name so that we can get telemetry that shows that this fix was invoked. Unfortunately there's no better scheme for that yet.